bootcampspringboot/src/main/java/com/max/bootcampspringboot/service/SkillService.java

```diff
@@ -1,7 +1,43 @@
 package com.max.bootcampspringboot.service;
 
+import com.max.bootcampspringboot.entity.Skill;
+import com.max.bootcampspringboot.repository.SkillRepository;
+import jakarta.persistence.EntityNotFoundException;
 import org.springframework.stereotype.Service;
 
+import java.util.List;
+
 @Service
 public class SkillService {
+
+    private final SkillRepository skillRepository;
+
+    public SkillService(SkillRepository skillRepository) {
+        this.skillRepository = skillRepository;
+    }
+
+    public Skill getSkill(int id) {
+        return this.skillRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Entity not found with id: " + id));
+    }
+
+    public List<Skill> getAllSkills() {
+        return this.skillRepository.findAll();
+    }
+
+    public Skill createSkill(Skill skill) {
+        return this.skillRepository.save(skill);
+    }
+
+    public Skill updateSkill(Skill skill) {
+        return this.skillRepository.findById(skill.getId()).map(oldSkill -> {
```

Comment:
Du kannst doch theoretisch hier die getSkill Methode nutzen und das Objekt davon dann überschreiben, das Fehlerhandling ist ja schon in der Funktion drin :)